crates/api_models/src/payment_methods.rs

```diff
@@ -65,6 +65,12 @@ pub struct PaymentMethodCreate {
     #[cfg(feature = "payouts")]
     #[schema(value_type = Option<Wallet>)]
     pub wallet: Option<payouts::Wallet>,
+
+    /// For Client based calls
+    pub client_secret: Option<String>,
+
+    /// Payment method data to be passed
+    pub payment_method_data: Option<PaymentMethodData>,
```

Comment:
can we remove the already existing variants at root level? I think its safe to remove it as long as nobody is using this api. Or are u forcing client to pass this field in order to perform client based calls?

Reply:
We wanted to keep the existing interface the API unaltered, while allowing the client based calls. Hence added new fields. For client based calls, client secret has to be passed along with relevant `payment_method_data `